File: frontend/src/types/product.ts
export interface Product {
  id: number
  company_id: number
  category_id: number
  name: string
  description: string | null
  brand: string | null
  image_url: string | null
  active: boolean
  created_at: string
  updated_at: string
}

export interface ProductCreate {
  category_id: number
  name: string
  description: string | null
  brand: string | null
  image_url: string | null
}

File: backend/app/schemas/product.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ProductCreate(BaseModel):
    category_id: int
    name: str = Field(min_length=2, max_length=150)
    description: str | None = None
    brand: str | None = Field(default=None, max_length=100)
    image_url: str | None = Field(default=None, max_length=500)


class ProductResponse(BaseModel):
    id: int
    company_id: int
    category_id: int
    name: str
    description: str | None
    brand: str | None
    image_url: str | None
    active: bool
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)